src/requestManager.cpp

```diff
@@ -1105,41 +1145,74 @@ bool CRequestManager::MarkBlockAsReceived(const uint256 &hash, CNode *pnode)
                 }
             }
         }
-        // BUIP010 Xtreme Thinblocks: end section
-        if (state->vBlocksInFlight.begin() == itInFlight->second.second)
+
+        if (state->vBlocksInFlight.begin() == mapBlocksInFlight[hash][nodeid])
```

Comment:
given lines 1149 and 1154 both use the same object, I'd like to see it gotten via "find" first.  And as a bonus check for if validity.  If mapBlocksInFlight[hash] is nonexistent, this is going to create it, for the sole purpose of deleting the non-existent  [hash][nodeid] later.